### PR TITLE
Server side rendering

### PR DESCRIPTION
**Pull request by ductiletoaster**
Major changes to the build including server side rendering in development and major improvements to the over all build.

### CODE REVIEW

File: lib/app/configs/webpack/common.ts [hunk 1]
```diff
@@ -166,11 +134,13 @@ const performance: Options.Performance = {
  */
 const plugins: Plugin[] = removeEmpty([
     /**
-     * This is necessary to emit hot updates (currently CSS only):
+     * Fixes a known issue with cross-platform differences in file watchers,
+     * so that webpack doesn't loose file changes when watched files change rapidly
```
Comment (jimmynotjim) [inline lib/app/configs/webpack/common.ts]: Nit: `loose` -> `lose`

File: src/server/views/Html.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    return (
+        <React.Fragment>
+            {files}
+        </React.Fragment>
+    );
+};
+
+const Html = (props: HtmlProps) => {
+    return (
+        <html lang="en">
```
Comment (jimmynotjim) [inline src/server/views/Html.tsx]: May want to add a TODO to handle `lang` switching in the future.

File: src/server/views/Html.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            <head>
+                <meta charSet="utf-8" />
+                <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
+                <meta name="theme-color" content="#000000" />
+                <link rel="manifest" href="/manifest.json" />
+                <link rel="shortcut icon" href="/favicon.ico" />
+                {props.files && <StyleSheet hrefs={props.files.css}/>}
+                <title>React App</title>
```
Comment (jimmynotjim) [inline src/server/views/Html.tsx]: Should we make the title a prop we can update dynamically?

File: src/shared/components/pages/Home/Home.tsx [hunk 4]
```diff
@@ -14,7 +13,7 @@ class Home extends React.Component<RouteComponentProps> {
                         <Grid.Column>
                             <Segment inverted={true} vertical={true} textAlign="center">
                                 <Container>
-                                    <Image className="app-logo" src={"/public/static/media/logo.svg"} />
```
Comment (jimmynotjim) [inline src/shared/components/pages/Home/Home.tsx]: You'll want to revert this to keep the logo visible.

File: src/server/views/Html.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            </head>
+            <body>
+            <noscript>
```
Comment (jimmynotjim) [inline src/server/views/Html.tsx]: Is this still needed if everything's being rendered on the server?

File: src/server/views/Html.tsx (hunk 3 above)
Comment (ductiletoaster) [reply src/server/views/Html.tsx]: In the final version we will. This template will require a lot of work.

File: src/server/views/Html.tsx (hunk 2 above)
Comment (ductiletoaster) [reply src/server/views/Html.tsx]: Will do

File: src/server/views/Html.tsx (hunk 5 above)
Comment (ductiletoaster) [reply src/server/views/Html.tsx]: I copied this template from create-react-app. We will likely write our own soon but no we don't really need it.

**Pull request merged**